source/class/qx/ui/treevirtual/SimpleTreeDataModel.js

```diff
@@ -831,9 +831,16 @@ qx.Class.define("qx.ui.treevirtual.SimpleTreeDataModel", {
       // Set selections in the selection model now
       var selectionModel = this.getTree().getSelectionModel();
       var selections = this._selections;
-      for (var nodeId in selections) {
-        var nRowIndex = this.getRowFromNodeId(nodeId);
-        selectionModel.setSelectionInterval(nRowIndex, nRowIndex);
+
+      selectionModel.setBatchMode(true);
+      try {
+        selectionModel.resetSelection();
+        for (var nodeId in selections) {
+          var nRowIndex = this.getRowFromNodeId(nodeId);
+          selectionModel.addSelectionInterval(nRowIndex, nRowIndex);
+        }
+      } finally {
+        selectionModel.setBatchMode(false);
```

Comment:
Is it possible that batch mode was already `true` initially, and this code should retrieve the current batch mode before line 835, and then restore that prior batch mode here rather than always assuming it was `false`?

Reply:
@derrell , you're right, it could be true already. Someone has implemented reference counting in the batchMode which should suffice, I think?

Reply:
https://github.com/qooxdoo/qooxdoo/blob/84400c57ae4fce6dd01169b5a1b7adfe89b7ed1c/source/class/qx/ui/table/selection/Model.js#L140

Reply:
I suppose. That's hidden, rather than clean and obvious code -- it requires looking at that setter to know that -- so my inclination would be to save the current value and restore that value when I'm done. You're right, it's not strictly necessary. I'll approve and let you decide whether to make it obvious or leave it as a more hidden side-effect.